packages/vm/src/wasm_backend/gatekeeper.rs

```diff
@@ -95,460 +100,183 @@ impl FunctionMiddleware for FunctionGatekeeper {
-            | Operator::I32GeU
-            | Operator::I64Eqz
-            | Operator::I64Eq
-            | Operator::I64Ne
-            | Operator::I64LtS
-            | Operator::I64LtU
-            | Operator::I64GtS
-            | Operator::I64GtU
-            | Operator::I64LeS
-            | Operator::I64LeU
-            | Operator::I64GeS
-            | Operator::I64GeU
-            | Operator::I32Clz
-            | Operator::I32Ctz
-            | Operator::I32Popcnt
-            | Operator::I32Add
-            | Operator::I32Sub
-            | Operator::I32Mul
-            | Operator::I32DivS
-            | Operator::I32DivU
-            | Operator::I32RemS
-            | Operator::I32RemU
-            | Operator::I32And
-            | Operator::I32Or
-            | Operator::I32Xor
-            | Operator::I32Shl
-            | Operator::I32ShrS
-            | Operator::I32ShrU
-            | Operator::I32Rotl
-            | Operator::I32Rotr
-            | Operator::I64Clz
-            | Operator::I64Ctz
-            | Operator::I64Popcnt
-            | Operator::I64Add
-            | Operator::I64Sub
-            | Operator::I64Mul
-            | Operator::I64DivS
-            | Operator::I64DivU
-            | Operator::I64RemS
-            | Operator::I64RemU
-            | Operator::I64And
-            | Operator::I64Or
-            | Operator::I64Xor
-            | Operator::I64Shl
-            | Operator::I64ShrS
-            | Operator::I64ShrU
-            | Operator::I64Rotl
-            | Operator::I64Rotr
-            | Operator::I32WrapI64
-            // Those are part of the MVP
-            // https://github.com/bytecodealliance/wasm-tools/blob/wasmparser-0.107.0/crates/wasmparser/src/lib.rs#L287-L288
-            | Operator::I64ExtendI32S
-            | Operator::I64ExtendI32U
-            // Sign-extension
-            // https://github.com/bytecodealliance/wasm-tools/blob/wasmparser-0.107.0/crates/wasmparser/src/lib.rs#L307-L311
-            | Operator::I32Extend8S
-            | Operator::I32Extend16S
-            | Operator::I64Extend8S
-            | Operator::I64Extend16S
-            | Operator::I64Extend32S => {
-                state.push_operator(operator);
-                Ok(())
-            }
-            Operator::RefNull { .. }
-            | Operator::RefIsNull
-            | Operator::RefFunc { .. }
-            | Operator::ReturnCall { .. }
-            | Operator::ReturnCallIndirect { .. }
-            | Operator::TypedSelect { .. }
-            | Operator::TableFill { .. }
-            | Operator::TableGet { .. }
-            | Operator::TableSet { .. }
-            | Operator::TableGrow { .. }
-            | Operator::TableSize { .. }
-            | Operator::CallRef { .. }
-            | Operator::ReturnCallRef { .. }
-            | Operator::RefAsNonNull
-            | Operator::BrOnNull { .. }
-            | Operator::BrOnNonNull { .. } => {
+        struct ProposalValidator<'a, 'b> {
```

Comment:
Could we move `struct ProposalValidator` and `impl<'a, 'b> ProposalValidator` out of here to make `fn feed` more readable? Maybe just a private type in the same file